portfolio/src/main/webapp/script.js

```diff
@@ -55,10 +55,14 @@ function loadMusicVideo(){
 }
 
 /**
- * Fetches a message from the data servlet
+ * Fetches data from the data servlet
  */
-async function getMsgFromServlet(){
-  const response = await fetch('/data');
-  const msg = await response.text();
-  document.getElementById("servlet-msg-container").innerText = msg; 
+function getDataFromServlet(){
+ fetch('/data').then(response => response.json()).then((data) => { 
+     const dataElement = document.getElementById("servlet-msg-container");
+     dataElement.innerHTML = '';
+     dataElement.appendChild(document.createTextNode(" " + data[0]));
```

Comment:
What happens if there aren't precisely three elements? (For example if data is written by a user or you even make a mistake elsewhere). It would be better to do the appends in a for loop or use the join function to join them with " " as a separator.

Reply:
You're definitely right, I only wrote it that way since it was just as a tester for the walkthrough. The code for the comment section replaces the above and I have a loop there.